agent/config/parse_windows.go

```diff
@@ -32,8 +34,108 @@ const (
 	// domain join check validation. This is useful for integration and
 	// functional-tests but should not be set for any non-test use-case.
 	envSkipDomainJoinCheck = "ZZZ_SKIP_DOMAIN_JOIN_CHECK_NOT_SUPPORTED_IN_PRODUCTION"
+
+	ReleaseId2004SAC        = "2004"
+	ReleaseId1909SAC        = "1909"
+	WindowsServer2019       = "Windows Server 2019"
+	WindowsServer2016       = "Windows Server 2016"
+	WindowsServerDataCenter = "Windows Server Datacenter"
+	InstallationTypeCore    = "Server Core"
+	InstallationTypeFull    = "Server"
+	UnknownOSType           = "unknown"
+	osTypeFormat            = "WINDOWS_SERVER_%s_%s"
+	ecsWinRegistryRootKey   = registry.LOCAL_MACHINE
+	ecsWinRegistryRootPath  = `SOFTWARE\Microsoft\Windows NT\CurrentVersion`
 )
 
+var winRegistry dependencies.WindowsRegistry
+
+func init() {
+	winRegistry = dependencies.StdRegistry{}
+}
+
+func setWinRegistry(mockRegistry dependencies.WindowsRegistry) {
+	winRegistry = mockRegistry
+}
+
+func getInstallationType(installationType string) (string, error) {
+	if InstallationTypeFull == installationType {
+		return "FULL", nil
+	} else if InstallationTypeCore == installationType {
+		return "CORE", nil
+	} else {
+		err := seelog.Errorf("Unsupported Installation type:%s", installationType)
+		return "", err
+	}
+}
+
+func getReleaseIdForSACReleases(productName string) (string, error) {
+	if strings.HasPrefix(productName, WindowsServer2019) {
+		return "2019", nil
+	} else if strings.HasPrefix(productName, WindowsServer2016) {
+		return "2016", nil
+	}
+	err := seelog.Errorf("Unsupported productName:%s for Windows SAC Release", productName)
+	return "", err
+}
+
+func getReleaseIdForLTSCReleases(releaseId string) (string, error) {
+	if ReleaseId2004SAC == releaseId {
+		return ReleaseId2004SAC, nil
+	} else if ReleaseId1909SAC == releaseId {
+		return ReleaseId1909SAC, nil
+	}
+	err := seelog.Errorf("Unsupported ReleaseId:%s for Windows LTSC Release", releaseId)
+	return "", err
+}
+
+func GetOperatingSystemFamily() string {
```

Comment:
please add comments to exported method `// GetOperatingSystemFamily ... ` https://github.com/golang/go/wiki/CodeReviewComments#doc-comments.  Same for parse_unsupported.go and parse_linux.go

Reply:
Done, Thank you